src/Advection/flat_advective_fluxes.jl

```diff
@@ -3,27 +3,44 @@
+@inline inner_right_biased_interpolate_xᶠᵃᵃ(i, j, k, ::XFlatGrid, scheme, ψ, args...) = @inbounds ψ[i, j, k]
+@inline inner_right_biased_interpolate_yᵃᶠᵃ(i, j, k, ::YFlatGrid, scheme, ψ, args...) = @inbounds ψ[i, j, k]
+@inline inner_right_biased_interpolate_zᵃᵃᶠ(i, j, k, ::ZFlatGrid, scheme, ψ, args...) = @inbounds ψ[i, j, k]
 
-        @inline advective_momentum_flux_Vu(i, j, k, grid::AbstractGrid{FT, TX, Flat}, scheme::$SchemeType, V, u) where {FT, TX} = zero(grid)
-        @inline advective_momentum_flux_Vw(i, j, k, grid::AbstractGrid{FT, TX, Flat}, scheme::$SchemeType, V, w) where {FT, TX} = zero(grid)
+@inline inner_right_biased_interpolate_xᶠᵃᵃ(i, j, k, grid::XFlatGrid, scheme, ψ::Function, idx, loc, args...) = @inbounds ψ(i, j, k, grid, args...)
+@inline inner_right_biased_interpolate_yᵃᶠᵃ(i, j, k, grid::YFlatGrid, scheme, ψ::Function, idx, loc, args...) = @inbounds ψ(i, j, k, grid, args...)
+@inline inner_right_biased_interpolate_zᵃᵃᶠ(i, j, k, grid::ZFlatGrid, scheme, ψ::Function, idx, loc, args...) = @inbounds ψ(i, j, k, grid, args...)
 
-        @inline advective_momentum_flux_Wu(i, j, k, grid::AbstractGrid{FT, TX, TY, Flat}, scheme::$SchemeType, W, u) where {FT, TX, TY} = zero(grid)
-        @inline advective_momentum_flux_Wv(i, j, k, grid::AbstractGrid{FT, TX, TY, Flat}, scheme::$SchemeType, W, v) where {FT, TX, TY} = zero(grid)
-    end
-end
\ No newline at end of file
+@inline inner_left_biased_interpolate_xᶠᵃᵃ(i, j, k, ::XFlatGrid, scheme, ψ, args...) = @inbounds ψ[i, j, k]
+@inline inner_left_biased_interpolate_yᵃᶠᵃ(i, j, k, ::YFlatGrid, scheme, ψ, args...) = @inbounds ψ[i, j, k]
+@inline inner_left_biased_interpolate_zᵃᵃᶠ(i, j, k, ::ZFlatGrid, scheme, ψ, args...) = @inbounds ψ[i, j, k]
+
+@inline inner_left_biased_interpolate_xᶠᵃᵃ(i, j, k, grid::XFlatGrid, scheme, ψ::Function, idx, loc, args...) = @inbounds ψ(i, j, k, grid, args...)
+@inline inner_left_biased_interpolate_yᵃᶠᵃ(i, j, k, grid::YFlatGrid, scheme, ψ::Function, idx, loc, args...) = @inbounds ψ(i, j, k, grid, args...)
+@inline inner_left_biased_interpolate_zᵃᵃᶠ(i, j, k, grid::ZFlatGrid, scheme, ψ::Function, idx, loc, args...) = @inbounds ψ(i, j, k, grid, args...)
+
+@inline inner_symmetric_interpolate_xᶠᵃᵃ(i, j, k, ::XFlatGrid, scheme, ψ, args...) = @inbounds ψ[i, j, k]
+@inline inner_symmetric_interpolate_yᵃᶠᵃ(i, j, k, ::YFlatGrid, scheme, ψ, args...) = @inbounds ψ[i, j, k]
+@inline inner_symmetric_interpolate_zᵃᵃᶠ(i, j, k, ::ZFlatGrid, scheme, ψ, args...) = @inbounds ψ[i, j, k]
+
+@inline inner_symmetric_interpolate_xᶠᵃᵃ(i, j, k, grid::XFlatGrid, scheme, ψ::Function, idx, loc, args...) = @inbounds ψ(i, j, k, grid, args...)
+@inline inner_symmetric_interpolate_yᵃᶠᵃ(i, j, k, grid::YFlatGrid, scheme, ψ::Function, idx, loc, args...) = @inbounds ψ(i, j, k, grid, args...)
+@inline inner_symmetric_interpolate_zᵃᵃᶠ(i, j, k, grid::ZFlatGrid, scheme, ψ::Function, idx, loc, args...) = @inbounds ψ(i, j, k, grid, args...)
```

Comment:
```suggestion

Grids = [:XFlatGrid, :YFlatGrid, :ZFlatGrid, :XFlatGrid, :YFlatGrid, :ZFlatGrid]

for side in (:left_biased, :right_biased, :symmetric)
    for (dir, Grid) in zip([:xᶠᵃᵃ, :yᵃᶠᵃ, :zᵃᵃᶠ, :xᶜᵃᵃ, :yᵃᶜᵃ, :zᵃᵃᶜ], Grids)
        interp_function = Symbol(side, :_interpolate_, dir)
        @eval begin
            $interp_function(i, j, k, grid::$Grid, ψ, args...) = @inbounds ψ[i, j, k]
            $interp_function(i, j, k, grid::$Grid, ψ::Function, args...) = @inbounds ψ(i, j, k, grid, args...)
        end
    end
end
```
since we need a bunch of interpolates maybe metaprogramming is good for it

Reply:
Done